File: backend/routers/v1/knowledge.go
package v1

import (
	kbAPI "devops-platform/internal/modules/knowledge/api"
	"devops-platform/internal/middleware"

	"github.com/gin-gonic/gin"
)

func registerKnowledgeBase(r *gin.RouterGroup) {
	g := r.Group("/kb")
	queryPermission := middleware.RequirePermission("knowledge", "list")
	updatePermission := middleware.RequirePermission("knowledge", "update")

	// Categories
	g.GET("/categories", queryPermission, kbAPI.ListCategories)
	g.POST("/categories", updatePermission,
		middleware.SetAuditOperation("创建知识库分类"),
		kbAPI.CreateCategory)
	g.PUT("/categories/:id", updatePermission, kbAPI.UpdateCategory)
	g.DELETE("/categories/:id", updatePermission,
		middleware.SetAuditOperation("删除知识库分类"),
		kbAPI.DeleteCategory)

	// Articles
	g.GET("/articles", queryPermission, kbAPI.ListArticles)
	g.GET("/articles/:id", queryPermission, kbAPI.GetArticle)
	g.POST("/articles", updatePermission,
		middleware.SetAuditOperation("创建知识库文章"),
		kbAPI.CreateArticle)
	g.PUT("/articles/:id", updatePermission, kbAPI.UpdateArticle)
	g.DELETE("/articles/:id", updatePermission,
		middleware.SetAuditOperation("删除知识库文章"),
		kbAPI.DeleteArticle)
}

File: frontend/src/api/knowledge.js
import request from './request'

export const listCategories = () => request.get('/kb/categories')
export const createCategory = (data) => request.post('/kb/categories', data)
export const updateCategory = (id, data) => request.put(`/kb/categories/${id}`, data)
export const deleteCategory = (id) => request.delete(`/kb/categories/${id}`)

export const listArticles = (params) => request.get('/kb/articles', { params })
export const getArticle = (id) => request.get(`/kb/articles/${id}`)
export const createArticle = (data) => request.post('/kb/articles', data)
export const updateArticle = (id, data) => request.put(`/kb/articles/${id}`, data)
export const deleteArticle = (id) => request.delete(`/kb/articles/${id}`)
